src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/CodeFixes/SyntaxEditorBasedCodeFixProvider.FixAllProvider.cs

```diff
@@ -20,91 +16,35 @@ internal partial class SyntaxEditorBasedCodeFixProvider : CodeFixProvider
         /// subclass needs to provide is how each document will apply all the fixes to all the 
         /// diagnostics in that document.
         /// </summary>
-        internal sealed class SyntaxEditorBasedFixAllProvider : FixAllProvider
+        internal sealed class SyntaxEditorBasedFixAllProvider : AbstractDocumentBasedFixAllProvider
         {
             private readonly SyntaxEditorBasedCodeFixProvider _codeFixProvider;
 
             public SyntaxEditorBasedFixAllProvider(SyntaxEditorBasedCodeFixProvider codeFixProvider)
                 => _codeFixProvider = codeFixProvider;
 
-            public sealed override async Task<CodeAction?> GetFixAsync(FixAllContext fixAllContext)
```

Comment:
all this logic was fully deleted.  we now reuse the code from DocumentBasedFixAllProvider (including hte new logic to process things one project at a time).

Reply:
Thank you! I have thought about similar cleanup for a while (without the perf improvements that you included in your PR).